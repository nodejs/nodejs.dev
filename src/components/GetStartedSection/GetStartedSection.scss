.get-started-section {
  .cta-container {
    display: flex;
    padding-top: var(--space-160);
    padding-bottom: var(--space-80);
    justify-content: space-evenly;
    flex-wrap: wrap;

    .cta {
      display: flex;
      flex-direction: column;
      justify-content: center;
      text-align: center;
      width: 422px;

      .header {
        margin-bottom: var(--space-16);
      }

      .body-text {
        color: var(--black7);
      }

      .image {
        margin-bottom: var(--space-12);
      }
    }
  }
  .btn {
    margin: 0 auto;
    background: var(--color-fill-button);
    padding: var(--space-08) var(--space-20);
    border-radius: 4px;
    font-weight: var(--font-weight-semibold);
    width: max-content;
    height: auto;

    &:focus,
    &:hover {
      background: var(--color-fill-button);
    }
  }
}
