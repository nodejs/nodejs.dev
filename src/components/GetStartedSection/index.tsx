import { Link } from 'gatsby';
import React from 'react';
import './GetStartedSection.scss';
import beginnerGuideIllustration from '../../images/illustrations/beginners-guide-illustration.svg';
import doMoreWithNodeIllustration from '../../images/illustrations/do-more-illustration.svg';

interface Props {
  learnLinkText: string;
  beginnerGuideHeaderText: string;
  beginnerGuideBodyText: string;
  doMoreWithNodeHeaderText: string;
  doMoreWithNodeBodyText: string;
}

const GetStartedSection = ({
  learnLinkText,
  beginnerGuideBodyText,
  beginnerGuideHeaderText,
  doMoreWithNodeBodyText,
  doMoreWithNodeHeaderText,
}: Props): JSX.Element => (
  <section className="get-started-section">
    <div className="cta-container">
      <div className="cta">
        <img src={beginnerGuideIllustration} alt="" className="image" />
        <h2 className="header">{beginnerGuideHeaderText}</h2>
        <p className="body-text">{beginnerGuideBodyText}</p>
      </div>
      <div className="cta">
        <img src={doMoreWithNodeIllustration} alt="" className="image" />
        <h2 className="header">{doMoreWithNodeHeaderText}</h2>
        <p className="body-text">{doMoreWithNodeBodyText}</p>
      </div>
    </div>
    <Link to="/learn" className="btn circular-container">
      {learnLinkText}
    </Link>
  </section>
);

export default GetStartedSection;
