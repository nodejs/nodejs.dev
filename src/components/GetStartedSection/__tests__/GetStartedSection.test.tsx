import React from 'react';
import { render } from '@testing-library/react';
import GetStartedSection from '..';

describe('Tests for Footer component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <GetStartedSection
        beginnerGuideBodyText="Mock Text"
        beginnerGuideHeaderText="Mock Text"
        doMoreWithNodeBodyText="Mock Text"
        doMoreWithNodeHeaderText="Mock Text"
        learnLinkText="Mock Text"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
