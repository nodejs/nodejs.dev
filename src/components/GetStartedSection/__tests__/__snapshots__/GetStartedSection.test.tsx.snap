// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Footer component renders correctly 1`] = `
<div>
  <section
    class="get-started-section"
  >
    <div
      class="cta-container"
    >
      <div
        class="cta"
      >
        <img
          alt=""
          class="image"
          src="test-file-stub"
        />
        <h2
          class="header"
        >
          Mock Text
        </h2>
        <p
          class="body-text"
        >
          Mock Text
        </p>
      </div>
      <div
        class="cta"
      >
        <img
          alt=""
          class="image"
          src="test-file-stub"
        />
        <h2
          class="header"
        >
          Mock Text
        </h2>
        <p
          class="body-text"
        >
          Mock Text
        </p>
      </div>
    </div>
    <a
      class="btn circular-container"
      href="/learn"
    >
      Mock Text
    </a>
  </section>
</div>
`;
